### PR TITLE
5 - add I block

### PR DESCRIPTION
**Pull request by LukaszProkop**
@pkierat-epam I-shaped block ready for review

### CODE REVIEW

**Comment by TJB4epam** [comment]
I've just merged other changes. Make sure you have all you need and that this works after you merge the new master. GitHub for one reports that there are conflicts in the testng.xml file.

**Comment by LukaszProkop** [comment]
@TJB4epam  Done

File: src/test/resources/testng.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
         <groups>
             <run>
                 <include name="Block"/>
+                <include name="IBlock"/>
```
Comment (TJB4epam) [inline src/test/resources/testng.xml:12]: What should be the difference between "Block" and "IBlock" groups? When would I want to run one, but not another? When will I want to run both?

I have a feeling that all tests in the IBlock group should belong to a Block group. Furthermore I'd want an LGroup just for the L-Block. What say you?

Comment (TJB4epam) [reply src/test/resources/testng.xml:12]: Also - you owe us two minor version numbers Łukasz. One for this, one for LBlock, so I'd like that to be bumped up in the POM (it's no longer Tetris 0.1).

File: src/test/java/com/epam/prejap/tetris/block/BlockFeedTest.java [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 import org.testng.annotations.Test;
 
+import java.lang.reflect.Field;
```
Comment (TJB4epam) [inline src/test/java/com/epam/prejap/tetris/block/BlockFeedTest.java:5]: Nifty. Will have to look at it deeper later, cause now I'm rushing to a meeting. Please, focus on other changes I requested in the meantime.

**Review by TJB4epam — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/prejap/tetris/block/IBlockTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        Block block = new Block(iBlock) {};
+        byte[][] actual = block.image;
+
+        //then
+        assertEquals(actual, expected);
+
+    }
+
+    private static byte[][] getIBlock() throws IllegalAccessException {
+        Field[] fields = IBlock.class.getDeclaredFields();
```
[the remark this answers is not in the document]
Comment (TJB4epam) [reply src/test/java/com/epam/prejap/tetris/block/IBlockTest.java:52]: Well, it's nice to see Reflection API used, but... there's a package method for that already. :D So, nope. :-) Paweł :+1:

**Comment by TJB4epam** [comment]
Closing - reflection was a nice but in the end a redundant touch. Please correct what was required and do reopen. Thank you!

**Comment by LukaszProkop** [comment]
@pkierat-epam @TJB4epam All necessary corrections have been implemented and all the files are up to date.

**Comment by pkierat-epam** [comment]
Points: 19

**Pull request merged**